docs/source/extracting-specification.py

```diff
@@ -0,0 +1,60 @@
+# Requires python 3
+#
+# Reads the stored specification from a freshly created NWB file and exports it
+# into pretty printed JSON files.
+
+from datetime import datetime
+from dateutil.tz import tzlocal
+
+import os
+import io
+import json
+import sys
+
+import h5py
+
+from pynwb import NWBFile, NWBHDF5IO
+
+
+def createNWBFile():
+
+    nwbfile = NWBFile('my first synthetic recording',
+                      'EXAMPLE_ID',
+                      datetime.now(tzlocal()))
+
+    filename = 'specification-extraction.nwb'
+
+    with NWBHDF5IO(filename, 'w') as nwbio:
+        nwbio.write(nwbfile, cache_spec=True)
+
+    return filename
+
+
+def exportSpecification(filename):
```

Comment:
Rather than making this part of the docs, I think it would be useful to have a function either in the HDF5IO backend (or maybe the [YAMLSpecReader ](https://github.com/hdmf-dev/hdmf/blob/8ddc8574123853215a4d6c43d27bd2aec908017d/src/hdmf/spec/namespace.py#L184) to load the specification(s)), which I guess would return either a ``NamespaceCatalog`` in case you load all namespaces of a ``SpecNamespace`` if you load a particular namespace from the file.

This would have the advantage of making the functionality more reusable, part of the regular tests, and also greatly simplify the tutorial.

Reply:
@oruebel Hmmpf. I'm not very inclined to add something I use at most once year into hdmf with tests.

How about adding this script, and the forgotten one in `docs/code` to the example tests? I did that in 3b152bbb (run_example_tests(): Test the scripts in docs/code as well, 2019-06-11).